internal/lb/client.go

```diff
@@ -0,0 +1,111 @@
+package lb
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const (
+	mediaType        = "application/json"
+	registerEndpoint = "/api/v1/nodes"
+)
+
+// RegisterRequest contains data needed to connect daemon with lb
+type RegisterRequest struct {
+	ID            string `json:"id"`
+	ConfigHash    string `json:"config_hash"`
+	NodeURL       string `json:"node_url"`
+	PayoutAddress string `json:"payout_address"`
+}
+
+// TokenResponse from lb register endpoint
+type TokenResponse struct {
+	Token string `json:"token"`
+}
+
+// Client used to communicate with vedran load balancer
+type Client struct {
+	client  *http.Client
+	BaseURL *url.URL
```

Comment:
url.Parse doesn't return error for invalid url
Will need to check that differently

Reply:
Checked and should be fine
NewRequest still returns req and do fails as it should